code/datums/brain_damage/mrat.dm

```diff
@@ -0,0 +1,63 @@
+/datum/brain_trauma/special/imaginary_friend/mrat
+	name = "Epistemania"
+	desc = "Patient suffers from a manic pursuit of knowlewdge."
+	scan_desc = "epistemania"
+	gain_text = "<span class='notice'>Requesting mentor rat...</span>"
+	lose_text = "<span class='warning'>Mentor rat not found. Either your mentor rat left you or you didn't find one.</span>"
+	resilience = TRAUMA_RESILIENCE_ABSOLUTE
+
+/datum/brain_trauma/special/imaginary_friend/mrat/get_ghost()
+	set waitfor = FALSE
+	var/list/mob/dead/observer/candidates = pollMentorCandidatesForMob("Do you want to play as [owner]'s mentor rat?", ROLE_PAI, null, null, 75, friend, POLL_IGNORE_IMAGINARYFRIEND)
+	if(LAZYLEN(candidates))
+		var/mob/dead/observer/C = pick(candidates)
+		friend.key = C.key
+		friend_initialized = TRUE
+		to_chat(owner, "<span class='notice'>You have acquired the mentor rat known as [friend.key], ask them any question you like.</span>")
+	else
```

Comment:
Probably need some kind of response if no mentors accept

Reply:
if no mentors accept it triggers lose_text
`lose_text = "<span class='warning'>Mentor rat not found. Either your mentor rat left you or you didn't find one.</span>"`

Reply:
Ah fair, didn’t see that